lesson6/hometask_frameq.rb

```diff
@@ -0,0 +1,87 @@
+require 'YAML'
+
+class Person
+
+  attr_accessor :age, :sex, :fullname, :file, :name
+
+  def self.add(name, &block)
+    self.class.send(:attr_accessor, name)
+    instance_variable_set("@#{name.to_sym}", Person.new(name, &block))
+  end
+
+  def initialize(name, &block)
+    @name = name
+    instance_eval(&block)
+    file if @file.nil?
+    @info = {:name => @name, :fullname => @fullname, :age => @age, :sex => @sex, :file => @file}
+    save!
+  end
+
+  def sex(sex = nil)
+    @sex = sex unless sex.nil?
+    @sex
+  end
+
+  def age(age = nil)
+    @age = age unless age.nil?
+    @age
+  end
+
+  def fullname(fullname = nil)
+    @fullname = fullname unless fullname.nil?
+    @fullname
+  end
+
+  def file(file = nil)
+    file.nil? ?  @file = "#{@name}.yaml" : @file = file
+  end
+
+  def inspect
+    @info
+  end
+
+  def save!
+    data = YAML.dump(@info)
```

Comment:
You can save just object itself, it will be pretty serialized/deserialized. I am not against your approach, just mention about such posibility.